py/parse_and_notify.py

```diff
@@ -54,6 +55,68 @@ def make_row(row: Sequence[str | int | None]) -> str:
     return f"%%%\n{table}%%%"
 
 
+def make_slack_message(
+    headers: Sequence[str],
+    content: Sequence[Sequence[str | int | None]],
+    error: bool,
+    error_message: str | None,
+) -> str:
+    """
+    Formats an ASCII table for slack message since slack's
+    markdown support is lacking
+    """
+    if not headers and not content:
+        return ""
+
+    def make_row(
+        row: Sequence[str | int | None], max_width: Sequence[int]
+    ) -> str:
+        content = "|".join(
+            (
+                " " + str(col).ljust(max_width[i]) + " "
+                for i, col in enumerate(row)
+            )
+        )
+        return f"|{content}|\n"
+
+    assert all(
+        len(row) == len(headers) for row in content
+    ), "Invalid table format."
+    # get max width of columns
+    num_cols = len(headers)
+    max_width = [len(h) for h in headers]
+    for i in range(num_cols):
+        for row in content:
+            max_width[i] = max(max_width[i], len(str(row[i])))
+
+    # 3 * (len(headers) - 1) because we need to add 3 dashes for each internal
+    # column separator (i.e ' | ')
+    line = ["-" * (sum(max_width) + 3 * (len(headers) - 1))]
+    rows = [make_row(r, max_width) for r in content]
+    table = (
+        f"```{make_row(headers, max_width)}"
+        + f"{make_row(line, max_width)}"
+        + f"{''.join(rows)}```"
+    )
```

Comment:
IT seems to me all this code will just work fine in the markdown version. Did you try to generate the markdown table this way as well? You could remove the copy paste.